source/utils/path.cpp

```diff
@@ -231,8 +229,20 @@ const std::string &path::string() const
 #ifdef _MSC_VER
 std::wstring path::wstring() const
 {
-    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> convert;
-    return convert.from_bytes(string());
+    const std::string &path_str = string();
+    int size = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, path_str.c_str(), static_cast<int>(path_str.length()), nullptr, 0);
+
+    if (size > 0)
+    {
+        std::wstring path_converted(size, L'\0');
+        // Note: const_cast is NOT necessary in C++17 and newer!
+        size = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, path_str.c_str(), static_cast<int>(path_str.length()), const_cast<wchar_t *>(path_converted.data()), size);
```

Comment:
1. Alternative to `const_cast` is to use `&path_converted[0]`. This is also what is used in the Microsoft doc: https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar#code-example

2. no need to store the return arg (or should we perform some error checking?).

Reply:
> 1. Alternative to `const_cast` is to use `&path_converted[0]`. This is also what is used in the Microsoft doc: https://learn.microsoft.com/en-us/windows/win32/api/stringapiset/nf-stringapiset-multibytetowidechar#code-example

True, I think this is actually the nicer solution in this case. Overall, I prefer using `data()` in such cases - however, until C++17 where [`data()` became non-const](https://en.cppreference.com/w/cpp/string/basic_string/data), `c_str()` and `data()` were equivalent. I prefer using `data()` rather than `&path_converted[0]` because it shows more exactly, in my opinion, what it does (access to the underlying raw data). However, I do agree that for pre-C++17 compilers, the `const_cast` is super ugly (and theoretically undefined behavior), so I'll change it :+1: 

>     2. no need to store the return arg (or should we perform some error checking?).

Agreed - I kept it there in case for debugging purposes, just in case anything goes wrong. Not that I assume that, though - I'm using these functions in some projects and they always worked really well. In the end, I think it neither makes the code worse, nor really better (unless there would be some hypothetical problem). I would leave it in because why not - but I don't mind throwing out the `size =` part either - as you wish! :wink:

Reply:
>  I would leave it in because why not

It may emit some compiler warning in the future, something like "assigned value is never used". But until a compiler actually does this, you can just leave it as it is (if you like it for debugging purposes).

Reply:
My newest commit should solve both your points (of course, in release builds, the variable is still not used, but as you said, it doesn't cause issues at the moment). Let me know if you have any other suggestions! :slightly_smiling_face: